tools/lsp/preview.rs

```diff
@@ -1182,12 +1210,13 @@ async fn reload_timer_function() {
                 cache.loading_state = PreviewFutureState::Pending;
                 break;
             }
-            PreviewFutureState::Pending => unreachable!(),
-            PreviewFutureState::PreLoading => unreachable!(),
             PreviewFutureState::NeedsReload => {
                 cache.loading_state = PreviewFutureState::PreLoading;
                 continue;
             }
+            PreviewFutureState::Pending
+            | PreviewFutureState::PreLoading
+            | PreviewFutureState::Configuring(_) => return,
```

Comment:
The code before was asserting that this state cannot be there. I think it is valuable to keep a panic for state that are clearly a bug. It tell the viewer of the code that this is impossible, and if this bug happen we get a report.

Reply:
II changed the state machine a bit, so that I have better state invariants now and can bring back those unreachables.